autotest/utilities/test_gdal_translate_lib.py

```diff
@@ -1158,6 +1158,57 @@ def test_gdal_translate_lib_dict_arguments():
     ]
 
 
+###############################################################################
+# Test -dmo option
+
+
+def test_gdal_translate_dmo_option():
+
+    dst_vrt = "/vsimem/test_dmo.vrt"
+
+    ## string dmo input
+    ds = gdal.Translate(
+        dst_vrt,
+        "../gcore/data/byte.tif",
+        domainMetadataOptions="NEW_DOMAIN:META-TAG=value",
+    )
+
+    md = ds.GetMetadata("NEW_DOMAIN")
+    assert "META-TAG" in md, "Did not get META-TAG"
+
+    ds = None
+
+    ## list dmo input
+    ds = gdal.Translate(
+        dst_vrt,
+        "../gcore/data/byte.tif",
+        domainMetadataOptions=[
+            "NEW_DOMAIN:META-TAG=value",
+            "ANOTHER_NEW_DOMAIN:META2-TAG=value",
+        ],
+    )
+
+    md = ds.GetMetadata("ANOTHER_NEW_DOMAIN")
+    assert "META2-TAG" in md, "Did not get META2-TAG"
+
+    ds = None
+
+    ## pythonic dict dmo input
+    items = {
+        "domain_name": {"key": "value", "key1": "value1"},
+        "domain_2": {"key2": "value2"},
+    }
+    ds = gdal.Translate(
+        "/vsimem/dmo.vrt",
+        gdal.Open("../gcore/data/byte.tif"),
+        domainMetadataOptions=items,
+    )
+    md = ds.GetMetadata("domain_2")
+    assert "key2" in md, "Did not get key2"
+
+    ds = None
+
```

Comment:
It would be good to add tests for invalid value of domainMetadataOptions, like "FOO" and "FOO=BAR"